pkg/kubelet/kuberuntime/security_context_windows.go

```diff
@@ -0,0 +1,55 @@
+// +build windows
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"fmt"
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/pkg/securitycontext"
+)
+
+var (
+	windowsRootUserName = "ContainerAdministrator"
+)
+
+// verifyRunAsNonRoot verifies RunAsNonRoot on windows.
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/20190103-windows-node-support.md#v1container
+// Windows does not have a root user, we cannot judge the root identity of the windows container by the way to judge the root(uid=0) of the linux container.
+// According to the discussion of sig-windows, at present, we assume that ContainerAdministrator is the windows container root user,
+// and then optimize this logic according to the best time.
+// https://docs.google.com/document/d/1Tjxzjjuy4SQsFSUVXZbvqVb64hjNAG5CQX8bK7Yda9w
+func verifyRunAsNonRoot(pod *v1.Pod, container *v1.Container, uid *int64, username string) error {
+	effectiveSc := securitycontext.DetermineEffectiveSecurityContext(pod, container)
+	// If the option is not set, or if running as root is allowed, return nil.
+	if effectiveSc == nil || effectiveSc.RunAsNonRoot == nil || !*effectiveSc.RunAsNonRoot {
+		return nil
+	}
+	if effectiveSc.WindowsOptions != nil {
+		if effectiveSc.WindowsOptions.RunAsUserName != nil {
+			if *effectiveSc.WindowsOptions.RunAsUserName == windowsRootUserName {
+				return fmt.Errorf("container's runAsUser (%s) which will be regarded as root identity and will be break non-root policy", username)
+			}
+		}
```

Comment:
linux version of this method will return nil here. Meaning that if `WindowsOptions.RunAsUserName` was set to a value that is not root, it will OK that independently from values of `uid ` and `username `. Why is this one different?

Reply:
Already fixed
The logic of linux container is to judge by the security context of pod and container first, if the user does not input, judge by mirror metadata. Windows reuses this logic